From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Adam Farden <adam@farden.cz>
Date: Thu, 9 Aug 2018 09:26:47 +0200
Subject: [PATCH] Add msm8976 tasha sound card detection to msm8916 HAL

msm8976 is a variant of msm8952, both use the msm8916 HAL

Test: manual - build ok. Sound ok

Change-Id: I9ac11e781cf627fa5efe586c96e48bfd04f32485
Signed-off-by: Adam Farden <adam@farden.cz>

post_proc: Enable post processing for msm8952

Test: manual - build ok. sound ok

Change-Id:  Idc5ab339bb9c898205986ba0b4c7cc91febf19de
Signed-off-by: David Viteri <davidteri91@gmail.com>
Signed-off-by: Tim <tomgus1.2@gmail.com>
---
 hal/msm8916/hw_info.c | 8 +++++---
 post_proc/Android.mk  | 4 ++--
 2 files changed, 7 insertions(+), 5 deletions(-)

diff --git a/hal/msm8916/hw_info.c b/hal/msm8916/hw_info.c
index 0afb5104f43a377fb9fbe75ab617e70d1fc536d0..095d0529141180d937687a4f215300962126a7c6 100644
--- a/hal/msm8916/hw_info.c
+++ b/hal/msm8916/hw_info.c
@@ -54,8 +54,10 @@ static void update_hardware_info_8x16(struct hardware_info *hw_info, const char
         strlcpy(hw_info->name, "msm8952", sizeof(hw_info->name));
     }  else if (!strcmp(snd_card_name, "msm8952-l9300-snd-card")) {
         strlcpy(hw_info->name, "msm8952", sizeof(hw_info->name));
+    } else if (!strcmp(snd_card_name, "msm8976-tasha-snd-card")) {
+        strlcpy(hw_info->name, "msm8976", sizeof(hw_info->name));
     } else {
-        ALOGW("%s: Not an  8x16/8909/8952 device", __func__);
+        ALOGW("%s: Not an 8x16/8909/8952/8976 device", __func__);
     }
 }
 
@@ -69,8 +71,8 @@ void *hw_info_init(const char *snd_card_name)
         return NULL;
     }
 
-    if (strstr(snd_card_name, "msm8x16") || strstr(snd_card_name, "msm8909")
-        || strstr(snd_card_name, "msm8952") ||
+    if (strstr(snd_card_name, "msm8x16") || strstr(snd_card_name, "msm8909") ||
+        strstr(snd_card_name, "msm8952") || strstr(snd_card_name, "msm8976") ||
         strstr(snd_card_name, "msm-bg-snd-card")) {
         ALOGV("8x16 - variant soundcard");
 
diff --git a/post_proc/Android.mk b/post_proc/Android.mk
index 9c658a34c68d5f09990a1fcc1f9591bfba1408ba..c800d8f890460c6ed4e67088d37d82c4d6c95a29 100644
--- a/post_proc/Android.mk
+++ b/post_proc/Android.mk
@@ -1,4 +1,4 @@
-ifneq ($(filter msm8974 msm8226 msm8084 msm8992 msm8994 msm8996 msm8909 msm8998 sdm845 sdm710 sm8150,$(TARGET_BOARD_PLATFORM)),)
+ifneq ($(filter msm8974 msm8226 msm8084 msm8992 msm8994 msm8996 msm8909 msm8952 msm8998 sdm845 sdm710 sm8150,$(TARGET_BOARD_PLATFORM)),)
 
 LOCAL_PATH:= $(call my-dir)
 
@@ -43,7 +43,7 @@ endif
 
 ################################################################################
 
-ifneq ($(filter msm8992 msm8994 msm8996 msm8909 msm8998 sdm845 sdm710 sm8150,$(TARGET_BOARD_PLATFORM)),)
+ifneq ($(filter msm8992 msm8994 msm8996 msm8909 msm8952 msm8998 sdm845 sdm710 sm8150,$(TARGET_BOARD_PLATFORM)),)
 
 include $(CLEAR_VARS)
 
